solidity/test/token/EverclearTokenBridge.t.sol

```diff
@@ -0,0 +1,552 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity ^0.8.22;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+import "forge-std/Test.sol";
+import {TransparentUpgradeableProxy} from "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {MockMailbox} from "../../contracts/mock/MockMailbox.sol";
+import {ERC20Test} from "../../contracts/test/ERC20Test.sol";
+import {TestPostDispatchHook} from "../../contracts/test/TestPostDispatchHook.sol";
+import {TypeCasts} from "../../contracts/libs/TypeCasts.sol";
+
+import {EverclearTokenBridge} from "../../contracts/token/bridge/EverclearTokenBridge.sol";
+import {IEverclearAdapter, IEverclear} from "../../contracts/interfaces/IEverclearAdapter.sol";
+import {Quote} from "../../contracts/interfaces/ITokenBridge.sol";
+
+/**
+ * @notice Mock implementation of IEverclearAdapter for testing
+ */
+contract MockEverclearAdapter is IEverclearAdapter {
```

Comment:
as @cmcewen points out, the mocks introduce potential for bugs that we dont instrument for coverage
we should consider adding a fork test of the actual implementation or pulling in their contracts as dependencies of our tests

Reply:
Added a fork test